test/unit/apk-utils-specs.js

```diff
@@ -491,4 +491,111 @@ describe('Apk-utils', () => {
       }
     });
   }));
+  describe('installOrUpgrade', withMocks({adb}, (mocks) => {
+    const pkgId = 'io.appium.settings';
+    const apkPath = '/path/to/my.apk';
+
+    it('should execute install if the package is not present', async () => {
+      mocks.adb.expects('getApkInfo').withExactArgs([apkPath]).once().returns({
```

Comment:
Why is `apkPath` expected in an array. The code above just passes the variable through to `getApkInfo`, and you're passing in a string, not an array. Same goes for all the rest of the mocks.